src/content/guides/efk-stack-app-oidc/index.md

```diff
@@ -0,0 +1,385 @@
+
+1. In Azure platform, navigate to the app-registration created in the previous section
+2. In the sidebar, click "Token configuration"
+3. Click "+Add groups claim"
+4. Select group types to include in the token (usually Security groups)
+5. In "Customize token properties by type", select "Group ID" for every type
+6. Click "Save"
+
+For reference on how to define groups' optional claims, see [Microsoft's official documentation](https://docs.microsoft.com/en-us/Azure/active-directory/develop/active-directory-optional-claims#configuring-groups-optional-claims).
+
+How to define and manage the required groups, as well as how to obtain the group IDs of those groups is out of scope for this guide. Check [Microsoft's documentation on Azure AD](https://docs.microsoft.com/en-us/Azure/devops/server/admin/setup-ad-groups?toc=%2FAzure%2Fdevops%2Forganizations%2Ftoc.json&bc=%2FAzure%2Fdevops%2Forganizations%2Fbreadcrumb%2Ftoc.json&view=Azure-devops-2020&viewFallbackFrom=Azure-devops)
+
+## Step 2: Configure opendistro-security plugin
+
+Opendistro-security is configured through secrets. These secrets need to be present in the app's namespace for the EFK configuration to deploy.
+
+In this section, we create the target namespace and deploy the `configSecret` named `opendistro-security-config`.
+
+Default namespace for the efk-stack-app is "efk-stack-app".
+
+### Create namespace
+
+To create the namespace on the tenant cluster, run:
+
+```bash=
+kubectl create namespace efk-stack-app
+```
+
+### Create configSecret
+
+1. Save the .yaml lines below to a file called `config.yml`. Point `openid_connect_url` to the correct URL.
+2. On the cluster run:
+
+```bash=
+kubectl create secret generic -n efk-stack-app opendistro-security-config --from-file=./config.yml
+```
+
+```yaml=config.yml
+---
+_meta:
+  type: "config"
+  config_version: 2
+
+config:
+  dynamic:
+    http:
+      anonymous_auth_enabled: false
+      xff:
+        enabled: false
+        internalProxies: ".*"
+        remoteIpHeader: "x-forwarded-for"
+    authc:
+      basic_internal_auth_domain:
+        description: "Authenticate via HTTP Basic against internal users database"
+        http_enabled: true
+        transport_enabled: true
+        order: 0
+        http_authenticator:
+          type: "basic"
+          challenge: false
+        authentication_backend:
+          type: "internal"
+
+      openid_auth_domain:
+        http_enabled: true
+        transport_enabled: true
+        order: 1
+        http_authenticator:
+          type: "openid"
+          challenge: false
+          config:
+            enable_ssl: false
+            verify_hostnames: false
+            roles_key: groups
+            subject_key: preferred_username
+            openid_connect_url: https://login.microsoftonline.com/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/v2.0/.well-known/openid-configuration
+        authentication_backend:
+          type: noop
+    authz: {}
+```
```

Comment:
I would reorder this slightly - the secret's yaml should be shown before telling them how to use it to create the secret.

```suggestion
1. Save the .yaml lines below to a file called `config.yml`. Point `openid_connect_url` to the correct URL.

```yaml=config.yml
---
_meta:
  type: "config"
  config_version: 2

config:
  dynamic:
    http:
      anonymous_auth_enabled: false
      xff:
        enabled: false
        internalProxies: ".*"
        remoteIpHeader: "x-forwarded-for"
    authc:
      basic_internal_auth_domain:
        description: "Authenticate via HTTP Basic against internal users database"
        http_enabled: true
        transport_enabled: true
        order: 0
        http_authenticator:
          type: "basic"
          challenge: false
        authentication_backend:
          type: "internal"

      openid_auth_domain:
        http_enabled: true
        transport_enabled: true
        order: 1
        http_authenticator:
          type: "openid"
          challenge: false
          config:
            enable_ssl: false
            verify_hostnames: false
            roles_key: groups
            subject_key: preferred_username
            openid_connect_url: https://login.microsoftonline.com/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/v2.0/.well-known/openid-configuration
        authentication_backend:
          type: noop
    authz: {}
```

2. On the cluster run:

```bash=
kubectl create secret generic -n efk-stack-app opendistro-security-config --from-file=./config.yml
```
```

Reply:
hmmm, github's suggestion formatting failed there. Basically - shift this _below_ the config.yml code block:


2. On the cluster run:

```bash=
kubectl create secret generic -n efk-stack-app opendistro-security-config --from-file=./config.yml
```